portals/new_admin/portals/client/src/sign-in.ts

```diff
@@ -0,0 +1,108 @@
+
+    const tokenEndpoint = 'https://dev-kw-oeodk.us.auth0.com/oauth/token';
+    const stsEndoint = 'https://da59-203-94-95-4.in.ngrok.io/api/am/sts/v1/oauth2/token';
+
+    if (!tokenEndpoint || tokenEndpoint.trim().length === 0) {
+        return Promise.reject(new Error("Invalid token endpoint found."));
+    }
+
+    const code = new URL(window.location.href).searchParams.get(AUTHORIZATION_CODE);
+
+    const body = [];
+    body.push(`client_id=${requestParams.clientId}`);
+    body.push(`code=${code}`);
+    body.push("grant_type=authorization_code");
+    body.push(`redirect_uri=https://localhost:4000/users`);
+    body.push(`code_verifier=${window.sessionStorage.getItem("code_verifier")}`);
+
+    return axios.post(tokenEndpoint, body.join("&"))
+        .then((response: any) => {
+            if (response.status !== 200) {
+                return Promise.reject(new Error("Invalid status code received in the token response: "
+                    + response.status));
+            }
+            window.sessionStorage.setItem("token", response.data.access_token);
+
+            const body = [];
+            body.push(`client_id=${requestParams.clientId}`);
+            body.push(`scope=apim:api_manage apim:subscription_manage apim:tier_manage apim:admin`);
+            body.push("grant_type=urn:ietf:params:oauth:grant-type:token-exchange");
+            body.push(`subject_token_type=urn:ietf:params:oauth:token-type:jwt`);
+            body.push(`requested_token_type=urn:ietf:params:oauth:token-type:jwt`);
+            body.push(`subject_token=${response.data.access_token}`);
+            body.push(`org_handle=organization`);
+            return axios.post(stsEndoint, body.join("&"))
+                .then((response: any) => {
+                    window.sessionStorage.setItem("exchanged_token", response.data.access_token);
+                    window.location.href = "https://localhost:4000/users";
+                });
+        }).catch((error: any) => {
+            return Promise.reject(error);
+        });
+};
+
+
+// export const login = () : any => {
+//     var codeVerifier = getCodeVerifier();
+
+//     Promise.resolve()
+//         .then(() => {
+//             return getCodeChallenge(codeVerifier)
+//         })
+//         .then(function (codeChallenge) {
+//             window.sessionStorage.setItem("code_verifier", codeVerifier);
+
+//             var args = new URLSearchParams({
+//                 client_id: "FbCSH23HybQMV9UlXJfeKHogAEHojzCO",
+//                 redirect_uri: "https://localhost:4000/#/publisher/token",
+//                 response_type: "code",
+//                 scope: "openid",
+//                 state: "RlZyVjlqYUpHTzltWC42c2FNRDRJT1JPfk1+TUFEa0RLb04yZldwYkpxVA==",
+//                 code_challenge: codeChallenge,
+//                 code_challenge_method: "S256"
+//             });
+//             window.location = "https://dev-kw-oeodk.us.auth0.com/authorize?" + args;
+//         })
+// }
```

Comment:
Shall we remove this commented code if not needed?

Reply:
These commented lines are not there in the latest file.